oak_restricted_kernel/src/simpleio.rs

```diff
@@ -16,26 +16,29 @@
 
 use crate::mm::Translator;
 use alloc::collections::VecDeque;
+use core::alloc::Allocator;
 use oak_simple_io::SimpleIo;
 use sev_guest::io::PortFactoryWrapper;
 use x86_64::VirtAddr;
 
 /// A communications channel using a simple IO device.
```

Comment:
Should we explain why we need the `Allocator` in a comment near the code, too?

IIUC: "can't just rely on the system allocator" and "custom allocator to ensure that the vectors are allocated in the memory area reserved for guest-host communication"